### PR TITLE
Bug 1762868: revert #1177 and fix common templates in MCs

### PR DESCRIPTION
**Pull request by runcom**
What #1177 had inadvertently regressed is that our MCs aren't rolled back anymore. This was sort of fine till the BZ came in, which exposed another bug we had. Basically, the code was adding duplicate and triplicates of common files all over. If you pair that with #1177, you'll get the bug in the title.

### CODE REVIEW

File: pkg/controller/template/template_controller.go [hunk 1]
```diff
@@ -403,9 +404,10 @@ func (ctrl *Controller) syncControllerConfig(key string) error {
 		if err := ctrl.syncRunningStatus(cfg); err != nil {
 			return err
 		}
-	} else {
-		glog.V(2).Info("ControllerConfig didn't change, skipping templates sync")
-		return nil
+	}
+
+	if cfg.Spec.Version != version.Hash {
+		return ctrl.syncFailingStatus(cfg, fmt.Errorf("refusing to retemplate with different ControllerConfig and MCC versions: %s != %s", cfg.Spec.Version, version.Hash))
```
Comment (cgwalters) [inline pkg/controller/template/template_controller.go]: OK, so if the MCO has updated but the MCC hasn't, the MCC will block until it's updated.  Makes sense.

**Comment by cgwalters** [comment]
And to help me understand - in order to fix the bug, we need to backport this all the way to 4.1, right?

**Comment by runcom** [comment]
> And to help me understand - in order to fix the bug, we need to backport this all the way to 4.1, right?

I think 4.1 can live w/o the controllerconfig version fix (as it will compare against the empty string and still be correct)

As for the other common template fix, yeah, need proper backport till 4.1 but not as urgent maybe

File: pkg/controller/template/template_controller.go (hunk 1 above)
Comment (runcom) [reply pkg/controller/template/template_controller.go]: this should guard also the case where, somehow, the new MCC is running with new templates but the controllerconfig is the old one and viceversa

**Comment by runcom** [comment]
Good to go now with the revert to #1177 which is also needed for 4.2 (#1182) @mrunalp PTAL

/cherrypick release-4.2

**Comment by runcom** [comment]
> Good to go now with the revert to #1177 which is also needed for 4.2 (#1182) @mrunalp PTAL
> 
> /cherrypick release-4.2

4.2 BZ is already here https://bugzilla.redhat.com/show_bug.cgi?id=1763205 for the cherry-pick

**Comment by runcom** [comment]
/hold cancel
/skip

Good to go

**Pull request merged**